velox/functions/prestosql/ArraySort.cpp

```diff
@@ -40,23 +40,52 @@ void applyComplexType(
   BufferPtr indices = allocateIndices(inputElements->size(), context.pool());
   vector_size_t* rawIndices = indices->asMutable<vector_size_t>();
 
+  // Allocate nulls buffer with default not nulls.
+  BufferPtr nulls = allocateNulls(inputElements->size(), context.pool());
+  auto* rawNulls = nulls->asMutable<uint64_t>();
+
   const CompareFlags flags{.nullsFirst = false, .ascending = true};
+  auto decodedIndices = decodedElements->indices();
+
   rows.applyToSelected([&](vector_size_t row) {
     const auto size = inputArray->sizeAt(row);
     const auto offset = inputArray->offsetAt(row);
-    for (auto i = offset; i < offset + size; ++i) {
-      rawIndices[i] = i;
+    vector_size_t next = offset;
+    vector_size_t last = offset + size - 1;
+    while (next <= last) {
```

Comment:
It is a bit difficult for me to follow this logic. I wonder if it can be simplified. It also seems to me the test is incorrect. Curious if you already tried running the Fuzzer and whether it succeeds.

```
    std::sort(
        rawIndices + offset,
        rawIndices + offset + size,
        [&](vector_size_t& a, vector_size_t& b) {
          bool aNull = decodedElements->isNullAt(a);
          bool bNull = decodedElements->isNullAt(b);
          if (aNull) {
            return false;
          }
          if (bNull) {
            return true;
          }
          return baseElementsVector->compare(
                     baseElementsVector,
                     decodedIndices[a],
                     decodedIndices[b],
                     flags) < 0;
        });
```